addon/ajax-request.js

```diff
@@ -148,6 +164,17 @@ export default class AjaxRequest {
   }
 
   /**
+   * @method _getFullHeadersHash
+   * @private
+   * @param {Object} headers
+   * @return {Object}
+   */
+  _getFullHeadersHash(headers) {
+    const classHeaders = get(this, 'headers') || {};
+    return merge(classHeaders, headers);
```

Comment:
Oh, is that how `merge` works? I thought that it just returned a merged version of the two, not mutate the first one and then change it.

Does it rreally mutate the service headers object, though?  `get(this, 'headers') will return an object, and without calling `.set` on the `headers` property, it shouldn't be changed, right?